runtime/near-vm-runner/src/tests/runtime_errors.rs

```diff
@@ -841,196 +665,97 @@ fn test_address_overflow() {
-                    HostError::GasExceeded,
-                ))),
-            );
-            let res = make_simple_contract_call_ex(
-                &code,
-                "main",
-                vm_kind,
-                Some(ProtocolFeature::FixContractLoadingCost.protocol_version()),
-                loading_gas,
-            );
-            gas_and_error_match(
-                res,
-                loading_gas,
-                Some(VMError::FunctionCallError(FunctionCallError::HostError(
-                    HostError::GasExceeded,
-                ))),
-            );
-        });
+        test_builder()
+            .wat(ALMOST_TRIVIAL_CONTRACT)
+            .gas(44115963)
+            .protocol_features(&[ProtocolFeature::FixContractLoadingCost])
+            .expects(&[
+                expect![[r#"
+                    VMOutcome: balance 4 storage_usage 12 return data None burnt gas 44115963 used gas 44115963
+                    Err: Exceeded the prepaid gas.
+                "#]],
+                expect![[r#"
+                    VMOutcome: balance 4 storage_usage 12 return data None burnt gas 44115963 used gas 44115963
+                    Err: Exceeded the prepaid gas.
+                "#]],
+            ]);
     }
 
     /// Executing with enough gas to finish loading but not to execute the full
     /// contract should have the same outcome before and after.
     #[test]
     fn test_fn_loading_gas_protocol_upgrade_exceed_executing() {
-        with_vm_variants(|vm_kind: VMKind| {
-            let (code, exec_gas) = almost_trivial_contract_and_exec_gas();
-            let loading_gas = prepaid_loading_gas(code.len());
-            let total_gas = exec_gas + loading_gas;
-
-            let res = make_simple_contract_call_ex(
-                &code,
-                "main",
-                vm_kind,
-                Some(ProtocolFeature::FixContractLoadingCost.protocol_version() - 1),
-                total_gas - 1,
-            );
-            gas_and_error_match(
-                res,
-                total_gas - 1,
-                Some(VMError::FunctionCallError(FunctionCallError::HostError(
-                    HostError::GasExceeded,
-                ))),
-            );
-            let res = make_simple_contract_call_ex(
-                &code,
-                "main",
-                vm_kind,
-                Some(ProtocolFeature::FixContractLoadingCost.protocol_version()),
-                total_gas - 1,
-            );
-            gas_and_error_match(
-                res,
-                total_gas - 1,
-                Some(VMError::FunctionCallError(FunctionCallError::HostError(
-                    HostError::GasExceeded,
-                ))),
-            );
-        });
+        test_builder()
+            .wat(ALMOST_TRIVIAL_CONTRACT)
+            .gas(47406986)
+            .protocol_features(&[ProtocolFeature::FixContractLoadingCost])
```

Comment:
Hm, the previous test was designed to keep checking `ProtocolFeature::FixContractLoadingCost.protocol_version() - 1`, just to make sure it does not change it's behaviour. This is relevant in two cases:

1. When we stabilise the feature: Make sure stabilisation does not accidentally break old behaviour.
2. Any future changes that may break old behaviour.

I think your update only checks the latest protocol version `ProtocolFeature::FixContractLoadingCost`. Probably it would be worth it to preserve the test coverage of this test.

Reply:
That's confusing, but I think this actually checks what you want it to. Note that there are *two* expectations for this test, one before the feature and one after. 

In other words, `protocol_features([feat])` adds tests at `feat - 1` and `PROTOCOL_VERSION` (which is the confusing bit i wanted to comment about).

Or have you already seen the two expects, and are asking about something else?

Reply:
Oh I see. I have seen that there are two things checked but I missed the `-1`.

Now that is really confusing, so when I add `protocol_features([TWO, THREE])` then the versions that are actually tested are `[1,2,infinite]`, right?

It seems this actually does what I want to be tested but I would not have expected `.protocol_features(...)` to work this way. But I guess a doc comment on that function would have resolved it. :)

Reply:
https://github.com/near/nearcore/pull/6922/commits/ac6b14900a295fc2ee98cae51b6aec69e4228e7d/ tries to make this a touch less confusing.